### ISSUE TITLE
TopHat2 (Bowtie2) hangs at Generating SAM header

### ISSUE THREAD

**Issue by sammyjava** [BenLangmead/bowtie2#19]
My understanding is that this concerns a Bowtie2 call in TopHat2; with essentially default parameters:
```
$ tophat -p 2 -r 56 /mnt/data/AGPv3/AGPv3 s_1_1_paired-trimmed.fq.gz s_1_2_paired-trimmed.fq.gz
```
the process stalls at "Generating SAM header", consuming 100% of one CPU but never (over several days) proceeding:
```
[2015-02-21 08:37:44] Beginning TopHat run (v2.0.13)
-----------------------------------------------
[2015-02-21 08:37:44] Checking for Bowtie
		  Bowtie version:	 2.2.4.0
[2015-02-21 08:37:44] Checking for Bowtie index files (genome)..
[2015-02-21 08:37:44] Checking for reference FASTA file
[2015-02-21 08:37:44] Generating SAM header for /mnt/data/AGPv3/AGPv3
```
I'm pretty sure it's stuck, not just taking a long time on this pretty fast Xeon machine. I'd appreciate any tips on what to alter to get it past this stage!

**Comment by BenLangmead** [comment]
Hard to tell whether this is a TopHat2 or Bowtie2 issue.  Is bowtie2 running when the hang happens?

Ben

**Comment by sammyjava** [comment]
Oh, yes, forgot to mention that ```bowtie-align-s``` is the process that's spinning.

**Comment by val-antonescu** [comment]
It would be very useful if we can reproduce this behavior. Is it possible to get a hold of the reads that are causing this? Or is there a particular detail about the data, like repeats, length, etc that might make this data an exceptional case over the others? Any info is appreciated. 

thanks,
Val

**Comment by BenLangmead** [comment]
Any help you could give us to allow us to reproduce would be very helpful.
Best,
Ben

**Comment by BenLangmead** [comment]
Will close for now, but reopen if you can provide more details.
Ben

**Comment by wgk51** [comment]
I am having this error while trying to run tophat on a cluster. It spends hours generating the SAM header prior to aborting. Based on others' output, this step should be relatively fast. 

tophat -G $HOME/mm9/mm09.gtf $HOME/mm9/mm9_mrna_index 1_CON_1.fastq.gz 1_CON_2.fastq.gz

[2017-07-28 09:21:18] Beginning TopHat run (v2.1.0)
-----------------------------------------------
[2017-07-28 09:21:18] Checking for Bowtie
                  Bowtie version:        2.2.6.0
[2017-07-28 09:21:18] Checking for Bowtie index files (genome)..
[2017-07-28 09:21:18] Checking for reference FASTA file
[2017-07-28 09:21:18] Generating SAM header for $HOME/mm9/mm9_mrna_index

The test files worked just fine, so I think TopHat is properly installed. I have confirmed that the index/gtf files are where I directed and I am running tophat from the directory containing my fastq files. I have tried a few small changes, but always get stuck at this step. Any advice?
